packages/shared/src/components/post/PostActions.tsx

```diff
@@ -77,6 +81,31 @@ export function PostActions({
     await toggleDownvote({ payload: post, origin });
   };
 
+  const [getPost, updatePost] = useUpdateQuery(
+    postByIdQueryOptions({
+      id: post.id,
+    }),
+  );
+
+  useQuerySubscription(({ variables }) => {
+    const { type, entityId } = variables;
+
+    if (type === 'POST') {
+      if (entityId !== post.id) {
+        return;
+      }
+
+      const newPost = getPost();
+      newPost.post.userState = {
+        ...newPost.post.userState,
+        awarded: true,
+      };
+      newPost.post.numAwards = (newPost.post.numAwards || 0) + 1;
+
+      updatePost(newPost);
```

Comment:
We use `getPost` to get immutable copy of current data for the post query (with structuredClone) which we can directly mutate and then use `updatePost` to update the query without using `queryClient`. Again all typed to avoid mistakes:
<img width="430" alt="image" src="https://github.com/user-attachments/assets/63cd9969-9ebf-44a9-a1ce-88e8c32ba7e9" />